### PR TITLE
Motive -> Motif

### PR DESCRIPTION
**Pull request by Sollace**
I realise @Mojang calls them "Motives" but that's incorrect. The feature on a painting is a _motif_, correcting that is my _motive_ for creating this PR.

### CODE REVIEW

**Comment by Juuxel** [comment]
They're registered in a registry called `minecraft:motive`.

**Comment by Sollace** [comment]
> 
> 
> They're registered in a registry called `minecraft:motive`.

Yeah, but it's still wrong.

https://www.dictionary.com/browse/motif

**Comment by Sollace** [comment]
From the definition of Motif:
image

**Comment by Juuxel** [comment]
We don't rename other bad registered names either, like `Entity.AREA_EFFECT_CLOUD` or `Decorator.HELL_FIRE`. I don't see why registries and the objects registered in them should be any different.

**Comment by Sollace** [comment]
> 
> 
> We don't rename other bad registered names either, like `Entity.AREA_EFFECT_CLOUD` or `Decorator.HELL_FIRE`. I don't see why registries and the objects registered in them should be any different.

Difference it that this one is clearly a typo. You wouldn't leave a field named `Entity.ARAE_EFFECT_CLUOD` in, would you?

**Comment by Fuyukai** [comment]
There's also EntityPaintingMotiveFix. It's clearly deliberate.

**Comment by Sollace** [comment]
> 
> 
> There's also EntityPaintingMotiveFix. It's clearly deliberate.

The choice here is either we use the name Mojang gave it, or we use a name that actually describes the object.

**Comment by i509VCB** [comment]
I agree with the name change since a motive sounds like a cause.

Motif sounds like the correct name for the object even though the registry is incorrectly named. 

Now this isn't an apple -> orange rename, the name is still related to the internal name of the registry.

**Comment by Earthcomputer** [comment]
I was very confused initially when I saw the "motive" registry, I would have been less confused if it were called "motif". Plus, if we use the argument that we should call it "motive" just because Mojang calls it that, means that we should also rename "container" to "menu". Except this is worse than the container/menu debate, as a motif is a real and well defined English word meaning exactly this, and motive means something else and is confusing.

**Comment by Prospector** [comment]
Motive is a synonym for Motif in this context. Not a necessary change. Motive is just as correct as Motif.

image

File: mappings/net/minecraft/entity/decoration/AbstractDecorationEntity.mapping [hunk 1]
```diff
@@ -1,12 +1,19 @@
 CLASS net/minecraft/class_1530 net/minecraft/entity/decoration/AbstractDecorationEntity
+	FIELD field_7097 obstructionCheckCoutner I
```
Comment (liach) [inline mappings/net/minecraft/entity/decoration/AbstractDecorationEntity.mapping]:
```suggestion
	FIELD field_7097 obstructionCheckCounter I
```

**Review by liach — CHANGES_REQUESTED** [review]

**Review by Prospector — COMMENTED** [review]
This is like voting to change DyeColor to DyeColour. No point, stick with the mojang name since it's just as correct.

**Comment by i509VCB** [comment]
> This is like voting to change DyeColor to DyeColour. No point, stick with the mojang name since it's just as correct.

That is American vs British English in that case.

Motive, similarly to a criminal's motive invokes the idea of an action/cause rather than a representation of an idea (which a motif is).

**Comment by Prospector** [comment]
see my other message. The second definition for motive is literally "a motif"

**Comment by Earthcomputer** [comment]
As a native English speaker, I did not know that "motive" was a synonym of "motif". Maybe I'm weird and just haven't seen it before, but either way there's ambiguity in "motive", and absolutely no ambiguity in "motif". That, along with the fact that "motive" is much more commonly used to mean something else, is a good reason for this change.

**Comment by Prospector** [comment]
Honestly I hadn't heard it either until I saw it in the registry name either but once I looked it up it immediately made sense. I don't think people not knowing a real word is a valid reason to change a name. It's not a name that could possibly be confused with something else. If there was another concept in Minecraft that involved the other definition of motive then I may feel differently. To me this is exactly like the issue with people not understanding what Clientbound/Serverbound mean. I don't think a lack of understanding of English should dictate the direction yarn moves in.

The worst case scenario with the current name is that someone is confused for a moment but then learns a new word. Best case scenario is they already know the word.

**Comment by Prospector** [comment]
That said, I wouldn't be entirely opposed to naming it to like PAINTING_MOTIVE or something just to clarify. However, there's no reason that a mod couldn't expand the motives to something other than paintings if they wanted to so perhaps that is not a great idea after a second thought.

**Comment by Earthcomputer** [comment]
> It's not a name that could possibly be confused with something else

Oh, you'd be hilariously surprised, when I first saw it, I thought the default entry "kebab" was the motive for like a crime, and I thought the registry was just Mojang's test of the registry system, and someone being funny. It's incredible that I actually thought that but it just goes to show how badly known the "motive" synonym is.

Of course I would be in favour of PAINTING_MOTIVE or PAINTING as well as MOTIF, those remove the confusion.

**Comment by liach** [comment]
https://www.dictionary.com/browse/motive

See definition 3 under the noun definitions (you need to click "SEE MORE")

image

**Comment by Sollace** [comment]
> Motive is a synonym for Motif in this context. Not a necessary change. Motive is just as correct as Motif.

I'm siding with @liach here. I write as a side-hobby, and generally have better knowledge of english than the layman, and even _I_ didn't know that Motive could be used a synonym for Motif, and that's most likely because people (Americans) who don't know how to spell hear motif and write motive meaning the former.

The synonym meaning is _incredibly_ obscure, almost more obscure than the word "motif" itself, so nothing changes here.

**Comment by modmuss50** [comment]
I’ve never encountered the word motif before

**Comment by Sollace** [comment]
> 
> 
> > It's not a name that could possibly be confused with something else
> 
> Oh, you'd be hilariously surprised, when I first saw it, I thought the default entry "kebab" was the motive for like a crime, and I thought the registry was just Mojang's test of the registry system, and someone being funny. It's incredible that I actually thought that but it just goes to show how badly known the "motive" synonym is.
> 
> Of course I would be in favour of PAINTING_MOTIVE or PAINTING as well as MOTIF, those remove the confusion.

Looking back, I can probably recount a similar experience to what @Earthcomputer is describing. The first time I saw it I was looking at the registries and thought "MOTIVE? Is this something to do with entity AI or something?"

I didn't bother looking looking into the usage, because it wasn't what I was looking for, but I remember coming out of it believing the registry must have been some kind of test of the registry system by Mojang or something.

**Comment by liach** [comment]
So motive is not a wrong term. We should add javadoc to all of these indicating the word "motive" means like "motif" to clarify. That's why we have javadocs.

**Comment by Sollace** [comment]
> 
> 
> So motive is not a wrong term. We should add javadoc to all of these indicating the word "motive" means like "motif" to clarify. That's why we have javadocs.

It's not _technically_ wrong in the same way I'm _technically_ a monkey.

**Comment by Earthcomputer** [comment]
"Motive" is not technically wrong, apparently, but it's a horribly confusing name. It's different from clientbound/serverbound in that literally every native English speaker in this conversation has had to consult a dictionary. Literally the _only_ argument for keeping it as "motive" is that that's what Mojang calls them, and we haven't exactly followed that reasoning in the past, have we?

What I think might be quite likely is that someone from Mojang knew the word in their native language and consulted a bilingual dictionary to translate it into English, and picked the wrong synonym, without knowledge of what a native English speaker would have chosen to call it. Yarn names shouldn't be bound by this, they should be the least confusing names for English speakers.

Motive is confusing, motif is apparently confusing for some, fine. How about renaming it to "painting", then? Or "painting_motif"?

**Comment by Prospector** [comment]
let's go with PAINTING_MOTIVE so you know what it means and you get to learn a new word.

**Comment by Sollace** [comment]
> 
> 
> let's go with PAINTING_MOTIVE so you know what it means and you get to learn a new word.

You mean `PAINTING_MOTIF` as @Earthcomputer suggested?

**Comment by Sollace** [comment]
We can't exactly call it a painting, because we have `PaintingEntity`. How does a painting have a painting? That doesn't make much sense.

You can say a painting has an _image_. But a painting having a painting is kind of redundant.

**Comment by Prospector** [comment]
No, keep the mojang name just add a clarifying prefix to it since motive has two definitions.

**Comment by Sollace** [comment]
No. The first thing everyone thinks of when they read motive is of a murder.

It's a bad name and shouldn't be involved. At all.

**Comment by Prospector** [comment]
It is just as valid of a word. With a painting prefix there is 0 confusion, and if anyone wasn't aware of the other definition of motive they will learn something.

**Comment by Sollace** [comment]
I would much rather people learn the definition of Motif than learn the wrong definition of Motive. It makes no difference either way, either, because in both cases you have people reaching for a dictionary.

(or possibly not, because they think they know what Motive means, in which case you'll be getting questions of how/why a painting can have a motive)

**Comment by Prospector** [comment]
Why are you so hung up on the fact that a word can mean multiple things? It is factually, objectively, **not wrong**. Therefore, it won't be changed.

**Comment by coderbot16** [comment]
`PAINTING_MOTIVE` would perfectly express that it's for paintings, clearing any confusion.

**Comment by Sollace** [comment]
It's an extremely obscure meaning, and (very clearly) confusing for a lot of people, and I don't think prefixing with 'PAINTING` will entirely clear that up. (if any it's going to raise more questions)

**Comment by coderbot16** [comment]
But motif is just as obscure. Might as well use the obscure word that the registry names use instead of creating another mismatch.

Also, all of the confusion in the first place appears to be in determining that it's related to paintings.

**Comment by Sollace** [comment]
The reason I prefer Motif over Motive is because people aren't going to automatically assume they know what it means and go straight to the first (or second) definition when it's actually used for the third definition.

**Comment by Sollace** [comment]
Anyway, I've argued this enough. You can choose to merge it or not. It won't affect me (as I have never worked with paintings in Minecraft and might not for a while)

**Comment by Sollace** [comment]
A wise man once said:
image

**Comment by liach** [comment]
My points against this rename:
1. Motive is a correct term.
2. The clarification should be made in comments (javadoc)

Other non-rename related changes, including some mappings for paintings, may go into another pr.

**Review by Earthcomputer — APPROVED** [review]

**Comment by Earthcomputer** [comment]
I have no real preference between `PAINTING_MOTIVE` vs `PAINTING_MOTIF`, as the prefix gives all the clarification needed. `MOTIF` on its own is better than `MOTIVE` on its own due to the difference in obscurity.

On a recent PR, I had named some methods with an "activate" and "deactivate" prefix, and liach suggested that I rename them to "enable" and "disable", which are synonyms, but it was for clarity. It's the same thing here, just because both words are obscure and that you may not have heard of either, doesn't change the fact that one is far more obscure than the other.

With that in mind, the only reason you have left to object to this change is because it's the same as the Mojang name. In which case we should merge the menu PR.

**Review by modmuss50 — APPROVED** [review]
After some thought, I dont see why not. It seems motif is the technically correct name here.

Its a little used part of the game (for mods) so I see no issue in changing it as it wont cause a headace for everyone.

**Comment by liach** [comment]
for the dfu one, i suggest keeping motive name and clarify in jd as motive is used in ldc string.

**Comment by Earthcomputer** [comment]
Javadocs don't exist to fix naming issues, they exist to add extra information about a good name. You wouldn't (or shouldn't) name loop variables "i" and "j" only to add a comment explaining what they mean, you'd rename "i" and "j" to something more meaningful. In this case, I would be in favour of adding a javadoc explaining what a "motif" is to those who don't know, while also doing the rename from "motive" to "motif" to help those who do know.
If I was not a native English speaker, I would prefer to learn a word that English speakers would actually use, than some obscure synonym that even native speakers don't understand.

**Review by i509VCB — APPROVED** [review]

**Comment by liach** [comment]
Motive is a valid word for "purpose/theme of a painting". As a result, I hereby oppose this change in where string literals refer to `motive`, e.g. in data fixer fix and registry code.

**Review by liach — APPROVED** [review]
Quality-wise this pull request is good, but I still hold by :-1: position on this rename proposal.

**Comment by Prospector** [comment]
Why is `painting_motive` not the best of all worlds? It both keeps the mojang name (which is *entirely* valid an objectively not incorrect) and also clarifies that it is for paintings as that is not clear with the pure mojang name.

**Comment by Earthcomputer** [comment]
+1 for renaming to `painting_motive`, while clarifying further that it's a "motif" in Javadocs.

**Comment by ghost** [comment]
Honestly I've never seen motif before so I'd get cofused when seeing this. Also I like prospectors proposal (painting_motive) better.

File: mappings/net/minecraft/entity/decoration/AbstractDecorationEntity.mapping (hunk 1 above)
Comment (NinjaPhenix) [inline mappings/net/minecraft/entity/decoration/AbstractDecorationEntity.mapping]: Is this a typo, `Coutner`

**Comment by Runemoro** [comment]
This doesn't make sense. Both motif and motive are correct spellings for the same term. Matching the in-game ID is more important than matching the usual spelling.

It's confusing to have to remember that for commands, you need to use "motive" while for code you need to use "motif".

**Comment by Prospector** [comment]
Yeah, I agree changing to motif makes no sense, though I do think adding a `painting` prefix to it could be a good thing. Unless we think the same motives could be expanded to other things?

**Comment by Earthcomputer** [comment]
This can be closed now since #1100 got merged.

**Pull request closed without merging**